### DISCUSSION TITLE
WAVEWATCH III NetCDF Compilation Issue

### DISCUSSION THREAD

**Discussion by krishnanathira** [NOAA-EMC/WW3#1033] [Q&A] [answered]
![2](https://github.com/NOAA-EMC/WW3/assets/109940994/b38c0495-329d-4b23-a5f2-0771d2e41683)
Hi 
I am trying to install WWIII 6.07 in the HPC of my Institute. I have set up all the required libraries (zlib, hdf5, netcdf, netcdf-c and netcdf fortran etc) first. I have specified the paths also with "export" command. Now while compilation I am getting an error **"ERROR: home/athirak/WW3/Libraries/grib2/bin/nc-config does not exist or is not executable"** 
But, the netcdf requirements are fullfilled. 
Attaching the screenshots.
Please support with possible help. 
![1](https://github.com/NOAA-EMC/WW3/assets/109940994/7717806d-9a79-48f7-a869-baf23d497cd9)

Thanks
Athira

**Comment by ukmo-ccbunney** [comment]
Hi @krishnanathira 
From your screen shot it looks like you have your path to nc_config set to your grib installation, not your netcdf installation?

Try running 'which no-config' to get the correct path.

Chris

**Reply by krishnanathira** [reply]
Thanks @ukmo-ccbunney for your help.

I have checked "which nc-config", it is showing ~/WW3/Libraries/grib2/bin/nc-config
I am adding the nc config -all and nf config -all. 

```
[athirak@pravah WW3]$ nc-config --all

This netCDF 4.8.0 has been built with the following features: 

  --cc            -> gcc
  --cflags        -> -I/home/athirak/WW3/Libraries/grib2/include -I/home/athirak/WW3/Libraries/grib2 -I/home/athirak/WW3/Libraries/grib2/include
  --libs          -> -L/home/athirak/WW3/Libraries/grib2/lib -lnetcdf
  --static        -> -lhdf5_hl -lhdf5 -lm -lcurl -L/home/athirak/WW3/Libraries/grib2/lib

  --has-c++       -> no
  --cxx           -> 

  --has-c++4      -> yes
  --cxx4          -> g++
  --cxx4flags     -> -I/home/athirak/WW3/Libraries/grib2/include -I/home/athirak/WW3/Libraries/grib2/include
  --cxx4libs      -> -L/home/athirak/WW3/Libraries/grib2/lib -lnetcdf_c++4 -lnetcdf

  --has-fortran   -> yes
  --fc            -> gfortran
  --fflags        -> -I/home/athirak/WW3/Libraries/grib2/include -I/home/athirak/WW3/Libraries/grib2/include
  --flibs         -> -L/home/athirak/WW3/Libraries/grib2/lib -lnetcdff
  --has-f90       -> 
  --has-f03       -> yes

  --has-dap       -> yes
  --has-dap2      -> yes
  --has-dap4      -> yes
  --has-nc2       -> yes
  --has-nc4       -> yes
  --has-hdf5      -> yes
  --has-hdf4      -> no
  --has-logging   -> no
  --has-pnetcdf   -> no
  --has-szlib     -> no
  --has-cdf5      -> yes
  --has-parallel4 -> no
  --has-parallel  -> no
  --has-nczarr    -> no

  --prefix        -> /home/athirak/WW3/Libraries/grib2
  --includedir    -> /home/athirak/WW3/Libraries/grib2/include
  --libdir        -> /home/athirak/WW3/Libraries/grib2/lib
  --version       -> netCDF 4.8.0


```
[athirak@pravah WW3]$ nf-config --all
```

This netCDF-Fortran 4.5.3 has been built with the following features: 

  --cc        -> gcc
  --cflags    -> -I/home/athirak/WW3/Libraries/grib2/include -I/home/athirak/WW3/Libraries/grib2/include

  --fc        -> gfortran
  --fflags    -> -I/home/athirak/WW3/Libraries/grib2/include -I/home/athirak/WW3/Libraries/grib2/include
  --flibs     -> -L/home/athirak/WW3/Libraries/grib2/lib -lnetcdff 
  --has-f90   -> 
  --has-f03   -> yes

  --has-nc2   -> yes
  --has-nc4   -> yes

  --prefix    -> /home/athirak/WW3/Libraries/grib2
  --includedir-> /home/athirak/WW3/Libraries/grib2/include
  --version   -> netCDF-Fortran 4.5.3
```

```

```

**Reply by ukmo-ccbunney** [reply]
Oh - well that is unexpected (to me)! 
What does it show if you `ls  ~/WW3/Libraries/grib2/bin/nc-config`?
Also, can you explicitly execute that on the command line, i.e. run `~/WW3/Libraries/grib2/bin/nc-config`?

**Reply by krishnanathira** [reply]
Yes.
ls  ~/WW3/Libraries/grib2/bin/nc-config shows
/home/athirak/WW3/Libraries/grib2/bin/nc-config

And I am not able to use the run command here.

Thanks

**Reply by ukmo-ccbunney** [reply]
> And I am not able to use the run command here.

Do you mean that running explicitly `/home/athirak/WW3/Libraries/grib2/bin/nc-config` on the command line doesn't work?

**Reply by ukmo-ccbunney** [reply, marked as answer]
Hold on - I've just spotted something.
In your original screenshot, you are missing a leading '/' on the path to your netCDF library:

<img width="478" alt="image" src="https://github.com/NOAA-EMC/WW3/assets/48915820/228d9845-97e7-471e-863c-ff6b0afc73b5">

That should be:
```
export NETCDF_HOME=/home/athirak/WW3/Libraries/grib2/bin/:${PATH}
```

That is most likely the issue! Sorry - I didn't spot it earlier.

**Comment by krishnanathira** [comment]
Thanks. I will try with that.

**Reply by krishnanathira** [reply]
Hi @ukmo-ccbunney 
I have reinstalled WWIII with mentioning all the netcdf paths correctly and it has compiled all the files and installed. Thanks for your help. 

Regards
Athira

**Reply by ukmo-ccbunney** [reply]
Great! Glad to hear it is working now!
